lib/spack/spack/util/path.py

```diff
@@ -60,6 +61,62 @@
 SPACK_PATH_PADDING_CHARS = '__spack_path_placeholder__'
 
 
+def is_path_url(path):
+    if '\\' in path:
+        return False
+    url_tuple = urlparse(path)
+    return bool(url_tuple.scheme) and url_tuple.scheme != 'c'
```

Comment:
Since we check for `\` in `path`, I don't think we need to check `url_tuple.scheme != 'c'` (the primary reason I bring this up is because this check would miss paths that include other drives on Windows, but that doesn't matter since the check at the beginning of the function will cover it - it should be removed though because IMO it is distracting).

Reply:
However, we may (and do in at least one place in Spack AFAIK) pass a string like `C;/user/bin/etc/example` which urlparse reduces to a tuple with `scheme == '<drive letter>'` which will read as a url according to `bool(url_tuple.scheme)`.
I could handle by verifying that the scheme is not a single character, AFAIK that wouldn't crop up in another instance.

Reply:
As I dig into this, I can't find any official means of recognizing a path as being a windows path. I agree that I cannot think of any case where scheme is a single letter (I assume this would always be a drive on Windows) so that seems reasonable as a check.

It might be more explicit to rule this out with a regular expression:

```
if re.match(path, r'[A-Z][a-z]\:'):
  return False
url_tuple = ...
```

As I think on it more, it occurs to me we might want to be careful about all "incoming" paths to Spack: try to convert these to `Path` objects (which `pathlib` would satisfy for Python > 3.4) but in any case I think that would make sense to do as a follow-up PR (after the main Windows support PR and this PR).

Reply:
I agree that converting to Path objects would be ideal, but best served as a later effort. Will implement the discussed changes.

Reply:
Actually the windows style filepaths are giving match a hard time, (try `re.match('C:\\<any string>', <any valid pattern>) which results in  an `re.error: bad escape (end of pattern) at position 2`) so I think we probably just change the scheme check.

Reply:
Done in new commit.